clients/gateway.go

```diff
@@ -105,12 +105,18 @@ type Transaction struct {
 	SenderAddress *felt.Felt `json:"sender_address"`
 }
 
+type TransactionFailureReason struct {
+	Code         string `json:"code"`
+	ErrorMessage string `json:"error_message,omitempty"`
+}
+
 type TransactionStatus struct {
-	Status           string       `json:"status"`
-	BlockHash        *felt.Felt   `json:"block_hash"`
-	BlockNumber      *big.Int     `json:"block_number"`
-	TransactionIndex *big.Int     `json:"transaction_index"`
-	Transaction      *Transaction `json:"transaction"`
+	Status                   string                    `json:"status"`
+	BlockHash                *felt.Felt                `json:"block_hash"`
+	BlockNumber              *big.Int                  `json:"block_number"`
+	TransactionIndex         *big.Int                  `json:"transaction_index"`
+	Transaction              *Transaction              `json:"transaction"`
+	TransactionFailureReason *TransactionFailureReason `json:"transaction_failure_reason"`
```

Comment:
Do you have an example URL that has this field in its response?

Reply:
yup. i only see it in the case of not enough fee at this point of time https://alpha4.starknet.io/feeder_gateway/get_transaction_receipt?transactionHash=0x2253fed5e39a8c695e282befa2e3fa67ef0dab656b71a36c32387d5c6030c4a 

i think after sierra, we will see more of this?